torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -253,7 +254,12 @@ c10::intrusive_ptr<OwnerRRef> RRefContext::getOrCreateOwnerRRef(
     // private.
     auto rref = c10::make_intrusive<OwnerRRef>(getWorkerId(), rrefId, type);
     owners_[rref->rrefId()] = rref;
-    ownerCV_.notify_all();
+    // ownerCV_.notify_all();
+    const auto pendingOwnerIter = pendingOwners_.find(rrefId);
```

Comment:
Why does `getOrCreateOwnerRRef` not have to add to `pendingOwners_`  if the pendingOwner is not found, but `getOwnerRRef` does?

Reply:
Because the code here creates the OwnerRRef, while `getOwnerRRef` does not have enough info to create the `OwnerRRef` and have to wait for `getOrCreateOwnerRRef` to create it. This traces back to where these functions are called. The `getOrCreateOwnerRRef` is called in `SCRIPT_REMOTE_CALL` and `PYTHON_REMOTE_CALL` request processors, which always create the `OwnerRRef`

This is a good catch. We should have renamed the function to `createOwnerRRef` and let me try if we can get rid of the retrieving part. This was originally called `getOrCreateOwnerRRef` back when both `rpc.remote()` and `rref.to_here()` can create `OwnerRRefs`. I think we can simplify that logic now.

Reply:
Cannot get rid of the retrieving logic from `getOrCreateOwnerRRef`, because unpickle can also create the RRef. So the main difference between `getOrCreateOwnerRRef` and `getOwnerRRef` is whether we have the TypePtr information. As long as TypePtr is available, we can create and do not need to return a future.

Reply:
Let me add above comments to those functions.

Reply:
Thanks for clarifying it!